packages/components/src/components/inline-message/inline-message.tsx

```diff
@@ -0,0 +1,51 @@
+import React from 'react';
+import { isMobile } from '@deriv/shared';
+import Icon from '../icon/icon';
+import Text from '../text';
+import './inline-message.scss';
+
+const type_to_icon_mapper = {
+    warning: 'IcAlertWarning',
+    information: 'IcAlertInfo',
+    announcement: 'IcAlertAnnounce',
+    error: 'IcAlertDanger',
+};
+
+const size_to_font_size_mapper = {
+    xs: isMobile() ? 'xxxxs' : 'xxxs',
+    sm: isMobile() ? 'xxxs' : 'xxs',
+    md: isMobile() ? 'xxs' : 'xs',
+    lg: isMobile() ? 'xs' : 's',
+};
+
+type TProps = {
+    type?: 'warning' | 'information' | 'announcement' | 'error';
+    size?: 'xs' | 'sm' | 'md' | 'lg';
+    title?: React.ReactNode;
+    message?: React.ReactNode;
+} & ({ title: React.ReactNode } | { message: React.ReactNode } | { children: React.ReactNode });
+
+const InlineMessage: React.FC<React.PropsWithChildren<TProps>> = ({
+    type = 'warning',
+    size = 'xs',
+    title,
+    message,
+    children,
+}) => {
+    const icon = type_to_icon_mapper[type];
+    const icon_size = size === 'lg' && !isMobile() ? 24 : 16;
```

Comment:
can we have a variable for the icon size in `packages/shared/src/styles/constants.scss` ? ( if it's a global variable which will be used in other places )

Reply:
@shayan-deriv We don't have any icon size there now, Also this is in JS so putting it in CSS doesn't make sense 🤔 Let's keep it as it is 🙏🏻